### DISCUSSION TITLE
Passing state via Next/link

### DISCUSSION THREAD

**Discussion by DiegoMcDipster** [vercel/next.js#21066] [Help] [answered]
When using the react-router-dom you can pass the state via Link.  For example:

// <Link  to={{ pathname: toUrl, state: {data}  }}>

What's the equivalent in NextJs?

**Comment by jamesmosier** [comment, marked as answer]
Hi @DiegoMcDipster. There is no equivalent in Next.js. You would either need to store the data in `localStorage` (or `sessionStorage`) yourself or save the state in a database and refetch it on the next page. Alternatively you could use a similar paradigm to the [`with-dynamic-app-layout`](https://github.com/vercel/next.js/blob/canary/examples/with-dynamic-app-layout/pages/index.js) example and in the layout set state in there and then pass it to the page components.

**Comment by DiegoMcDipster** [comment]
@jamesmosier , thanks for the reply and the advice on the alternative ways to do it!  I'm going with local storage.
